### PR TITLE
feat: asset tabs in project page based on routing

### PR DESCRIPTION
**Pull request by shawnyama**
# Description

https://user-images.githubusercontent.com/28237258/222170381-9d3c0800-0c54-49fb-82d1-64170342d8a6.mov

- Tabs are opened based on route
- Tabs are removed based on index
- Edited/simplified old tab components now `tera-tab-group.vue`
- Synced tabs and active index with tab storage but I think this can be done better. I want to see about getting `computed` are some equivalent working to keep track of the tab storage.

Resolves #(issue)

### CODE REVIEW

File: packages/client/hmi-client/src/types/common.ts [hunk 1]
```diff
@@ -79,6 +79,8 @@ export type SidePanelTab = {
 
 // Tabs
 export type Tab = {
-	name: string;
-	props?: Object;
+	label: string;
+	icon: string;
+	assetId?: string | number;
+	assetType?: string; // Change to AssetType later
```
Comment (YohannParis) [inline packages/client/hmi-client/src/types/common.ts:85]: Can't the `assetType` be inferred when fetching the _Asset_ in the _Tab_ component?

File: packages/client/hmi-client/src/page/project/components/tera-resource-sidebar.vue [hunk 2]
```diff
@@ -159,4 +178,17 @@ nav {
 	overflow: hidden;
 	white-space: nowrap;
 }
+
+.p-tree:deep(.p-treenode-content:has(span[active='true'])),
```
Comment (YohannParis) [inline packages/client/hmi-client/src/page/project/components/tera-resource-sidebar.vue:182]: Why the `:has()`? did a simple `.p-treenode-content [active=true]` did not work?
Or is that you want the `.p-treenode-content` to highlight when the `span[active=true]` exist?

File: packages/client/hmi-client/src/page/project/components/tera-resource-sidebar.vue [hunk 3]
```diff
@@ -14,24 +14,40 @@
...
+						route.params.assetType === slotProps.node.data.assetType
+					"
+				>
+					{{ slotProps.node.label }}
+					<Chip :label="slotProps.node.data.assetType" />
+				</span>
 			</template>
 		</Tree>
+		<div v-else class="loading-spinner">
+			<div><i class="pi pi-spin pi-spinner" style="font-size: 4rem" /></div>
```
Comment (YohannParis) [inline packages/client/hmi-client/src/page/project/components/tera-resource-sidebar.vue:36]: font-size should be in the `<style>`

**Review by YohannParis — DISMISSED** [review]
Nice! 
Please remove all commented code.

File: packages/client/hmi-client/src/page/project/tera-project.vue [hunk 4]
```diff
@@ -86,45 +85,80 @@ import Model from '@/components/models/Model.vue';
...
-			}
+// Associated with tab storage
+const tabContext = props.project?.id.toString();
+const openTabs = ref<Tab[]>([]);
+const activeTabIndex = ref(0);
+// const openTabs = computed(() => tabStore.getTabs(tabContext));
+// const activeTabIndex = computed(() => tabStore.getActiveTabIndex(tabContext));
+
+// @ts-ignore
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
```
[the remark this answers is not in the document]
Comment (mwdchang) [reply packages/client/hmi-client/src/page/project/tera-project.vue:124]: To add a bit more here, a lot of times you can either add underscores, e.g. `$subscribe((mutation, _state) > ...` to give hints that the variable is unused, or comment out the parameters like

```
$subscribe((/* mutation, state */) => ....
```

File: packages/client/hmi-client/src/page/project/components/tera-resource-sidebar.vue (hunk 2 above)
Comment (shawnyama) [reply packages/client/hmi-client/src/page/project/components/tera-resource-sidebar.vue:182]: I shortened the second rule but I still have to use the span. I use the span to trigger if it's active (you can see my custom template in the tree widget). Once span triggers I use the has to change the parent style

File: packages/client/hmi-client/src/page/project/tera-project.vue (hunk 4 above)
[the remark this answers is not in the document]
Comment (shawnyama) [reply packages/client/hmi-client/src/page/project/tera-project.vue]: I can't seem to find that file. I found `.eslintrc.cjs` it's not in there. Anyways I just commented the variables for now

File: packages/client/hmi-client/src/types/common.ts (hunk 1 above)
Comment (shawnyama) [reply packages/client/hmi-client/src/types/common.ts]: Not sure exactly what you mean. I made `assetType` have a type of `ProjectAssetType` but I am not sure if that's what you meant

**Pull request merged**